packages/x-date-pickers/src/internals/models/props/clock.ts

```diff
@@ -0,0 +1,59 @@
+import { SxProps, Theme } from '@mui/material/styles';
+import { BaseTimeValidationProps, TimeValidationProps } from '../../hooks/validation/models';
+import { PickerSelectionState } from '../../hooks/usePicker/usePickerValue';
+import { TimeView } from '../../../models';
+import type { ExportedDigitalClockProps } from '../../../DigitalClock/DigitalClock.types';
+import type { ExportedUseViewsOptions } from '../../hooks/useViews';
+
+export interface ExportedBaseClockProps<TDate>
+  extends TimeValidationProps<TDate>,
+    BaseTimeValidationProps {
+  /**
+   * 12h/24h view for hour selection clock.
+   * @default `utils.is12HourCycleInCurrentLocale()`
+   */
+  ampm?: boolean;
+}
+
+export interface BaseClockProps<TDate> extends ExportedUseViewsOptions<TimeView> {
+  className?: string;
+  /**
+   * The system prop that allows defining system overrides as well as additional CSS styles.
+   */
+  sx?: SxProps<Theme>;
+  /**
+   * The selected value.
+   * Used when the component is controlled.
+   */
+  value?: TDate | null;
+  /**
+   * The default selected value.
+   * Used when the component is not controlled.
+   */
+  defaultValue?: TDate | null;
+  /**
+   * Callback fired when the value changes.
+   * @template TDate
+   * @param {TDate | null} value The new value.
+   * @param {PickerSelectionState | undefined} selectionState Indicates if the date selection is complete.
+   */
+  onChange?: (value: TDate | null, selectionState?: PickerSelectionState) => void;
+  /**
+   * If `true`, the picker and text field are disabled.
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the picker and text field are readOnly.
+   * @default false
+   */
+  readOnly?: boolean;
+}
+
+export interface DesktopOnlyTimePickerProps<TDate> extends ExportedDigitalClockProps<TDate> {
+  /**
+   * Number of possible time options at which the single column time renderer is used.
+   * @default 24
+   */
+  renderTimeInASingleColumnThreshold?: number;
```

Comment:
That one is hard to name...

I'd be in favor of naming it in a way that feelds more "number" and less "render function"
Maybe `thresholdToRenderInASingleColumn`, or just `timeRenderedInASingleColumnThreshold` to avoid starting with `render`.


Or go one level lower with a `shouldRenderInASingleColumn?: (params: { optionCount: number }) => boolean`.

Reply:
Renamed it to [something in between](https://github.com/mui/mui-x/pull/7958/commits/84ddaaebe08cd3aae3ace5daab4ac8b92ec54dda), what do you think? 🤔